src/components/NewsCard/index.jsx

```diff
@@ -15,35 +14,35 @@ const NewsArticle = ({date, title, article}) => {
     )
 }
 
-const NewsCard = ({source, price, buttonText, details, ...otherProps}) => {
+const NewsCard = ({source, children, ...otherProps}) => {
     const boxColor = useColorModeValue("brand.100", "brand.700");
 
-    const {allNewsInfo} = useStockNews();
+    const {newsMasterlist} = useStockNews();
 
-    let newsDetails = allNewsInfo.filter(newsInfo => newsInfo.name === source)
-    let newsArticles;
+    let name;
+    let articles;
 
-    if (newsDetails.length === 1) {
-        newsArticles = allNewsInfo.filter(newsInfo => newsInfo.name === source)[0].articles
+    let targetSource = newsMasterlist.find(currSource => currSource.id === source.id)
+    if (targetSource !== undefined) {
+        name = targetSource.name;
+        articles = targetSource.articles;
     } else {
-        newsArticles = [];
+        name = "";
+        articles = [];
     }
 
-    //TODO: implement drag/drop and x icon
     return (
         <Box mx={3} mt={5} px={4} py={4} border="1px" borderColor="brand.400" borderRadius="lg" shadow="md" bg={boxColor} {...otherProps}>
-            {/*<Flex>*/}
-            {/*    <DragHandleIcon/>*/}
-            {/*    <Spacer/>*/}
-            {/*    <CloseIcon/>*/}
-            {/*</Flex>*/}
-            <Text align="center" mt="5px" fontSize="xl" fontWeight="bold">{source}</Text>
-
+            {children}
+            <Text align="center" mt="5px" fontSize="xl" fontWeight="bold" dhp>{name}</Text>
             <VStack align="flex-start">
                 {
-                    newsArticles.map(article => <NewsArticle date={article.date} title={article.title}
-                                                             article={article.src}/>)
-                }
+                    articles.map(article =>
```

Comment:
You can just destructure the article here 
articles.map(({date, title, src}) => .....}